xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/html/HTMLUtils.java

```diff
@@ -73,7 +73,7 @@
         /**
          * Regex to recognize a XML Entity.
          */
-        private static final Pattern ENTITY = Pattern.compile("&[a-zA-Z]+;|&#[0-9a-zA-Z]+;");
+        private static final Pattern ENTITY = Pattern.compile("&[a-zA-Z0-9]+;|&#[0-9a-zA-Z]+;");
```

Comment:
A documentation link to where this is coming from would be great. Isn't there some grammar that exists somewhere?

Reply:
Didn't find it yet. If you have an idea where to find it I take it.

Reply:
https://www.w3.org/TR/WD-xml-lang-970630#sec4.3 ?

Reply:
So for example `%_` seems valid but you don't seem to have it in your regex...

Reply:
Next question: is there some existing regex or API to check that?

Reply:
> My point is that we need to check against the grammar to be sure to not forget anything. 

Yes I agree that was part of my question when posting the sources :)